src/components/ChooseCurrency.tsx

```diff
@@ -0,0 +1,170 @@
+ *  @Currency
+ *  @param {string} label - should be in the format {Name} {ISO code}
+ *  @param {string} values - are uppercase ISO 4217 currency code
+ *  @param {string?} hasSymbol - if the currency has a symbol it should be represented as a string
+ *  @param {number} maxFractionalDigits - the standard fractional units used by the currency should be set with maxFractionalDigits
+ *
+ *  Bitcoin is represented as:
+ *  {
+ *      label: "bitcoin BTC",
+ *      value: "BTC",
+ *      hasSymbol: "₿",
+ *      maxFractionalDigits: 8
+ *  }
+ */
+export const FIAT_OPTIONS: Currency[] = [
+    {
+        label: "Bitcoin BTC",
+        value: "BTC",
+        hasSymbol: "₿",
+        maxFractionalDigits: 8
+    },
+    {
+        label: "United States Dollar USD",
+        value: "USD",
+        hasSymbol: "$",
+        maxFractionalDigits: 2
+    },
+    { label: "Swiss Franc CHF", value: "CHF", maxFractionalDigits: 2 },
+    {
+        label: "Chinese Yuan CNY",
+        value: "CNY",
+        hasSymbol: "¥",
+        maxFractionalDigits: 2
+    },
+    {
+        label: "Euro EUR",
+        value: "EUR",
+        hasSymbol: "€",
+        maxFractionalDigits: 2
+    },
+    {
+        label: "British Pound GBP",
+        value: "GBP",
+        hasSymbol: "₤",
+        maxFractionalDigits: 2
+    },
+    {
+        label: "Japanese Yen JPY",
+        value: "JPY",
+        hasSymbol: "¥",
+        maxFractionalDigits: 0
+    },
+    {
+        label: "Korean Won KRW",
+        value: "KRW",
+        hasSymbol: "₩",
+        maxFractionalDigits: 0
+    },
+    { label: "Kuwaiti Dinar KWD", value: "KWD", maxFractionalDigits: 3 }
+];
+
+export function ChooseCurrency() {
+    const i18n = useI18n();
+    const [error, setError] = createSignal<Error>();
+    const [state, _actions] = useMegaStore();
+    const [loading, setLoading] = createSignal(false);
+
+    function findCurrencyByValue(value: string) {
+        return FIAT_OPTIONS.find((currency) => currency.value === value);
+    }
+
+    const [_chooseCurrencyForm, { Form, Field }] =
+        createForm<ChooseCurrencyForm>({
+            initialValues: {
+                fiatCurrency: ""
+            },
+            validate: (values) => {
+                const errors: Record<string, string> = {};
+                if (values.fiatCurrency === undefined) {
+                    errors.fiatCurrency = i18n.t(
+                        "settings.currency.error_unsupported_currency"
+                    );
+                }
+                return errors;
+            }
+        });
+
+    const handleFormSubmit = async (f: ChooseCurrencyForm) => {
+        setLoading(true);
+        try {
+            localStorage.setItem(
+                "fiat_currency",
+                JSON.stringify(findCurrencyByValue(f.fiatCurrency))
+            );
+
+            await timeout(1000);
+            window.location.href = "/";
```

Comment:
is there a reason this requires a full reload rather than just a normal nav?

Reply:
I wasn't able to get actions.sync() to work to switch over to the new currency as a part of the async function

```
await actions.sync();
```
didn't really do anything so I just decided to reload from the homepage to force a sync

Reply:
I tried forcing a resync using
 
```
await actions.sync()
```

But nothing appeared to be happening, is there a better way to set a new state for the wallet?

Reply:
normally what I do is create a setter in `actions`. here's how I save an npub for example:
```
saveNpub(npub: string) {
            localStorage.setItem("npub", npub);
            setState({ npub });
}
```

so in this case there could be `saveFiat` method that saves the new npub chosen here.

and then there could be a `fetchPrice` method that `sync` and `saveFiat` both call

Reply:
moved this TODO in #524